### PR TITLE
fix(#major); opensea; add OpenSea v1 support

### PR DESCRIPTION
**Pull request by mrbrianhobo**
Subgraph Status: 
- OpenSea V1: https://okgraph.xyz/?q=messari%2Fopensea-v1-ethereum
- OpenSea V2: https://okgraph.xyz/?q=messari%2Fopensea-v2-ethereum

Updates:
- Restructure OpenSea subgraphs directory to contain both V1 and V2 (add deployment scripts for both based off of `uniswap-forks` dir)
- Add support for OpenSea V1 contract call handlers (calldata decoding for ERC721/ERC1155 `transferFrom`/`safeTransferFrom`)
- Add bundle sale logic (calldata decoding for `atomicize` in `WyvernAtomicizer`)
- Add Dutch Auction pricing 

NOTE: Revenue calculations for Collection entities are a bit weird due to multiple collection bundle sales/private sales (exact revenue metrics might be off)

### CODE REVIEW

File: subgraphs/opensea/protocols/opensea-v2/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const PROTOCOL_SUBGRAPH_VERSION = "1.0.0";
+export const PROTOCOL_METHODOLOGY_VERSION = "1.0.0";
+export const PROTOCOL_MARKETPLACE_NAME = "OpenSea: Wyvern Exchange v2";
```
Comment (this-username-is-taken) [inline subgraphs/opensea/protocols/opensea-v2/src/constants.ts:3]: I'd just call it `OpenSea v2`, which IMO is a more appropriate name for the marketplace

File: subgraphs/opensea/protocols/opensea-v2/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+export const PROTOCOL_MARKETPLACE_SLUG = "opensea-v2";
+export const PROTOCOL_MARKETPLACE_ADDRESS =
+  "0x7f268357A8c2552623316e2562D90e642bB538E5";
```
Comment (this-username-is-taken) [inline subgraphs/opensea/protocols/opensea-v2/src/constants.ts:6]: let's try keep all addresses lowercase going forward to avoid unexpected issues

**Review by this-username-is-taken — DISMISSED** [review]
Great work!

**Comment by 0xbe1** [comment]
@mrbrianhobo by bundle sale - are these items from the same collection, or can they come from multiple collections?

File: subgraphs/opensea/README.md [hunk 3]
```diff
@@ -245,21 +284,37 @@ The subgraph call handler flow can be broken down into several steps:
   - `0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2` for WETH
   - Subgraph does not track other currencies currently
 - `buy.basePrice`/`sell.basePrice` denotes base trade price
+  - Note that Dutch Auction sales use `calculateMatchPrice` to determine trade price
 - `buy.maker` denotes the NFT buyer (sale taker/bid maker)
+
+  Known Exceptions:
```
Comment (0xbe1) [inline subgraphs/opensea/README.md:290]: this is great docs!

**Review by 0xbe1 — APPROVED** [review]
decoding these callData bytes requires a gigabrain - glad we have gigabrian!

File: subgraphs/opensea/protocols/opensea-v2/src/constants.ts (hunk 2 above)
Comment (0xbe1) [reply subgraphs/opensea/protocols/opensea-v2/src/constants.ts:6]: a trick could be using `Address` type instead, which is case insensitive

**Comment by mrbrianhobo** [comment]
> @mrbrianhobo by bundle sale - are these items from the same collection, or can they come from multiple collections?

@0xbe1 you can have items from multiple collections and NFT types in a bundle sale
to be more specific, bundle sale `calldata` contains a list of target addresses and `calldata` for each target
- targets == ERC721/ERC1155 contracts
- calldata for each target == `transferFrom`/`safeTransferFrom` calls

example txn: https://etherscan.io/tx/0xe8629bfc57ab619a442f027c46d63e1f101bd934232405fa8e8eaf156bfca848
- contains ERC721 tokens from multiple collections
- contains ERC1155 token from a separate collection as well

i can probably add more supporting edge case transactions in the docs after
- bundle sales with different collections
- dutch action bundles
- gem/genie sweeps
- nifty gateway transactions

File: subgraphs/opensea/protocols/opensea-v2/src/constants.ts (hunk 2 above)
Comment (mrbrianhobo) [reply subgraphs/opensea/protocols/opensea-v2/src/constants.ts:6]: i'll just make it lowercase

File: subgraphs/opensea/README.md (hunk 3 above)
Comment (mrbrianhobo) [reply subgraphs/opensea/README.md:290]: thanks! i tried my best to clarify the subgraph call handler flow

File: README.md [hunk 4]
```diff
@@ -163,8 +163,8 @@ This repo contains subgraphs defined using a set of standardized schemas. These
 | ERC20 Tokens (2021) | 🛠 | 1.0.0 / 1.0.0 / 1.0.0 | [![ERC20 Ethereum](./docs/images/chains/ethereum.png)](https://thegraph.com/hosted-service/subgraph/messari/erc20-holders-2021) |
 | ERC20 Tokens (2022) | 🛠 | 1.0.0 / 1.0.0 / 1.0.0 | [![ERC20 Ethereum](./docs/images/chains/ethereum.png)](https://thegraph.com/hosted-service/subgraph/messari/erc20-holders-2022) |
 | **NFT Subgraphs** | | |
-| OpenSea (Wyvern v1) | |
-| OpenSea (Wyvern v2) | 🔨 |
+| OpenSea v1 (Wyvern Exchange) | 🛠 | 1.0.0 / 1.0.0 / 1.0.0 | [![OpenSea v1](./docs/images/chains/ethereum.png)](https://thegraph.com/hosted-service/subgraph/messari/opensea-v1-ethereum) |
```
Comment (this-username-is-taken) [inline README.md:166]: I'm assuming you'd want to increment the versions here?

**Review by this-username-is-taken — APPROVED** [review]
Great work! LGTM.

**Pull request merged**